### ISSUE TITLE
rook: Handling of test with unknown parameters in tests file 

### ISSUE THREAD

**Issue by mandd** [idaholab/SR2ML#38]
**Describe the bug**
If there is a miss-spelled or un-defined parameters used in the tests file, rook will treat it as invalid tester directory, see following: 

```
rook: found 10 test dirs under "/home/civet/civet/build_0/SR2ML/tests" ...
invalid tester directory: /home/civet/civet/build_0/SR2ML/tests/../src/Testers
Unknown: OrderedCsv
Unknown Parameters in: TestBasicEventScheduler /home/civet/civet/build_0/SR2ML/tests/tests
```

rook will now skip the tester folder. 

It will be nice if rook can error out especially in our regression test system, since the tests can be easily skipped due to miss spelled or undefined parameters. We will not know it until we checking carefully at the tests output. 

**To Reproduce**

**Expected behavior**
The regression machine should fail the test directly (as in the RAVEN regression suite)

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or task?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

**Comment by wangcj05** [comment]
@mandd I suggest to move this issue to RAVEN

**Comment by joshua-cogliati-inl** [comment]
Yes, this looks more like a RAVEN Rook issue.  When you open it there, can you attach a sample ```tests``` file that causes this?  Thanks.

**Comment by wangcj05** [comment]
Here is the simple example, if there is an unknown parameter in the rook, it will not error out: 

![image](https://user-images.githubusercontent.com/7321071/126519025-80e4066f-cd9d-4056-ba61-1dc4405baa05.png)

Here, OrderedCsv is not a valid parameter, the following is the rook message: 

```
rook: found 10 test dirs under "/home/civet/civet/build_0/SR2ML/tests" ...
invalid tester directory: /home/civet/civet/build_0/SR2ML/tests/../src/Testers
Unknown: OrderedCsv
Unknown Parameters in: TestBasicEventScheduler /home/civet/civet/build_0/SR2ML/tests/tests

```
